components/tidb_query_datatype/src/def/field_type.rs

```diff
@@ -289,7 +293,7 @@ pub trait FieldTypeAccessor {
         self.is_non_binary_string_like()
```

Comment:
BTW I think it might not be a good idea to introduce "RestoreData" concept to the collation. Collation itself does not have such concepts.